packages/amplify-codegen-e2e-tests/src/codegen-tests-base/test-setup.ts

```diff
@@ -0,0 +1,71 @@
+import {
+    deleteProjectDir,
+    deleteProject,
+    AmplifyFrontendConfig,
+    getAdminApp,
+    constructGraphQLConfig
+} from "amplify-codegen-e2e-core";
+import { existsSync, readFileSync } from "fs";
+import path from 'path';
+import { isNotEmptyDir, generateSourceCode } from '../utils';
+import { JSONUtilities } from 'amplify-cli-core';
+import { SandboxApp } from '../types/SandboxApp';
+import { load } from 'js-yaml';
+
+export const REGION = 'us-east-1';
+
+export function getGraphQLConfigFilePath(projectRoot: string): string {
+    return path.join(projectRoot, '.graphqlconfig.yml');
+}
+
+export async function deleteAmplifyProject(projectRoot: string) {
+    const metaFilePath = path.join(projectRoot, 'amplify', '#current-cloud-backend', 'amplify-meta.json');
+    if (existsSync(metaFilePath)) {
+    await deleteProject(projectRoot);
+    }
+    deleteProjectDir(projectRoot);
+}
+
+export function testSetupBeforeAddCodegen(projectRoot: string, config: AmplifyFrontendConfig): string {
+    // generate pre-existing user file
+    const userSourceCodePath = generateSourceCode(projectRoot, config.srcDir);
+
+    // verify that the pre-existing file exists and codegen output directory is empty
+    expect(existsSync(userSourceCodePath)).toBe(true);
+    expect(isNotEmptyDir(path.join(projectRoot, config.graphqlCodegenDir))).toBe(false);
```

Comment:
Before adding the codegen, this directory does not even exist rather than `isNotEmpty`.

Reply:
Changed this to test that the directory does not exist